src/main/scala/firrtl/passes/CheckWidths.scala

```diff
@@ -121,9 +121,13 @@ object CheckWidths extends Pass {
       // This is a leaf check of the "local" width-correctness of one expression node, so no recursion.
       expr match {
         case e @ UIntLiteral(v, w: IntWidth) if math.max(1, v.bitLength) > w.width =>
-          errors.append(new WidthTooSmall(info, target.serialize, v))
+          if (w.width > 0 || v != 0) { // UInt<0>(0) is allowed
```

Comment:
```suggestion
          if (w.width > 0 || (w.width ==0 && v != 0)) { // UInt<0>(0) is allowed
```
We don't want people trying to create negative width UInts! I know it won't parse, but one could construct the objects themselves in a transform and then call `CheckWidths`.

Reply:
done! thanks